### PR TITLE
add docs

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

add docs
**Loss Functions**

- ClassSimplexCriterion.md
- MSECriterion.md
- SoftMarginCriterion.md

**Non-linear Activations**

- SoftMax.md

**Normalization Layers**

- BatchNormalization.md

**Recurrent Layers**

- GRU.md

- RNN.md

**Simple Layers**

- Log.md

- Min.md

- MulConstant.md

- Reshape.md

**optim**

- LBFGS.md

### CODE REVIEW

File: docs/docs/APIdocs/layers/optim/LBFGS.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+**Scala:**
+```scala
+val optimMethod = new LBFGS(maxIter)
+```
+
+**Python:**
+```python
+val optim_method = LBFGS(maxIter, maxEval, tolFun, tolX, learningRate, lineSearch, lineSearchOptions)
```
Comment (zhichao-li) [inline docs/docs/APIdocs/layers/optim/LBFGS.md]: add an example here on how to integrated with optimizer?

File: docs/docs/APIdocs/layers/nn/Simple Layers/Reshape.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+**Python:**
+```python
+reshape = Reshape(size, batch_mode)
+```
+
+The `forward(input)` reshape the input tensor into `size(0) * size(1) * ...` tensor,
+taking the elements row-wise.
+
+**parameters**
+* **size** - the size after reshape
```
Comment (zhichao-li) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Reshape.md]: Not sure if we need to add the parameters details here as user can always get those from API doc. My concern is if we put those parameter descriptions here, it would be hard to sync with the java doc or python doc, and also the naming convention is diff between scala and python. @shane-huang

File: docs/docs/APIdocs/layers/nn/Simple Layers/Min.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+min = Min(2)
+input = np.array([
+  [1.0, 2.0],
+  [3.0, 4.0]
+])
+
+target = np.array([1.0, 1.0])
+output = min.forward(input)
+gradient = min.backward(input, target)
+-> print output
```
Comment (zhichao-li) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Min.md]: the other guys would print the result at the end of the example code instead of mixed between them. Anyway, I think it's acceptable written in this style. @shane-huang

File: docs/docs/APIdocs/layers/nn/Loss Functions/ClassSimplexCriterion.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+
+val criterion = ClassSimplexCriterion(5)
+val input = Tensor(T(
+ T(1.0f, 2.0f, 3.0f, 4.0f, 5.0f),
+ T(4.0f, 5.0f, 6.0f, 7.0f, 8.0f)
+))
+val target = Tensor(2)
+target(Array(1)) = 2.0f
+target(Array(2)) = 1.0f
```
Comment (JerryYanWan) [inline docs/docs/APIdocs/layers/nn/Loss Functions/ClassSimplexCriterion.md]: `target(1) = 2.0f target(2) = 1.0f`

File: docs/docs/APIdocs/layers/nn/Recurrent Layers/GRU.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  T(2.0f, 3.0f),
+  T(4.0f, 5.0f)
+)).resize(Array(1, seqLength, inputSize))
+val rec = Recurrent()
+
+val model = Sequential()
+    .add(rec.add(GRU(inputSize, hiddenSize)))
+    .add(TimeDistributed(Linear(hiddenSize, outputSize)))
+val output = model.forward(input)
+val gradient = model.backward(input, target)
```
Comment (JerryYanWan) [inline docs/docs/APIdocs/layers/nn/Recurrent Layers/GRU.md]: No criterion? Should the target be defined as gradOutput ?
